credentials/credentials-route.js

```diff
@@ -0,0 +1,233 @@
+const AWS = require('aws-sdk');
+var encryptor = require('file-encryptor');
+var fs= require('fs');
+const multer = require('multer');
+require("dotenv").config();
+const path = require('path');
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.REACT_APP_AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.REACT_APP_AWS_SECRET_ACCESS_KEY,
+  region: process.env.REACT_APP_AWS_REGION
+});
+console.log("access id", process.env.REACT_APP_AWS_ACCESS_KEY_ID)
+module.exports = (app) => {
+
+const uploadDirectory = './uploads';
+if (!fs.existsSync(uploadDirectory)) {
+  fs.mkdirSync(uploadDirectory);
+}
+  // Configure Multer storage
+const storage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, 'uploads/');
+  },
+  filename: (req, file, cb) => {
+    const fileName = file.originalname.toLowerCase().split(' ').join('-');
+    cb(null, fileName);
+  }
+});
+
+// Create a Multer instance
+const upload = multer({ storage });
+
+app.post('/upload', upload.single('file'), (req, res) => {
+  
+    try {
+      const file = req.file;
+      const fileName = req.body.fileName;
+      const extension = path.extname(fileName);
+      const filenameWithoutExtension = path.basename(fileName, extension);
+      const key = req.body.key;
+      
+
+      if (!file) {
+        return res.status(400).json({ error: 'No file provided' });
+      }
+
+      encryptor.encryptFile(req.file.path, filenameWithoutExtension+'-'+key+'.dat', key, function(err) {
+        const encryptedFile = fs.readFileSync('./'+filenameWithoutExtension+'-'+key+'.dat');
+        const params = {
+          Bucket: process.env.REACT_APP_BUCKET_NAME,
+          Key: filenameWithoutExtension+key,
+          Body: encryptedFile,
+          ACL: 'public-read',
+        };
+    
+        s3.upload(params, (err, data) => {
+          if (err) {
+            console.error(err);
+            res.status(500).json({ error: 'Failed to upload file to S3.' });
+          } else {
+            console.log('File uploaded successfully. ' + data.Location);
+            sendRequestEmail(req.body.providerEmail, key);
+            res.status(200).json({ message: 'File uploaded successfully.', location: data.Location });
+          }
+        });
+      });
+    } catch (err) {
+      console.error(err);
+      res.status(500).json({ error: 'An error occurred.' });
+    }
+  });
+
+  app.post('/download', async (req, res) => {
+
+    const fileName = req.body.fileName;
+    const extension = path.extname(fileName);
+    const filenameWithoutExtension = path.basename(fileName, extension);
+    const key = req.body.key;
+  
+    if (!fileName) {
+      return res.status(400).json({ error: 'File name is required' });
+    }
+ 
+    try {
+      const params = {
+        Bucket: process.env.REACT_APP_BUCKET_NAME, 
```

Comment:
is the bucket shared across all user personas?

Reply:
no, this is only for doctor credentials